code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -8,290 +9,140 @@
 	item_state = "electronic"
 	throw_speed = 3
 	throw_range = 7
-	materials = list(MAT_METAL=500)
-	var/obj/item/weapon/disk/nuclear/the_disk = null
-	var/active = 0
+	materials = list(MAT_METAL = 500, MAT_GLASS = 250)
+	var/active = FALSE
+	var/atom/movable/target = null //The thing we're searching for
+	var/nuke_warning = FALSE // If we've set off a miniature alarm about an armed nuke
+	var/mode = TRACK_NUKE_DISK //What are we looking for?
 
 /obj/item/weapon/pinpointer/New()
 	..()
 	pinpointer_list += src
 
 /obj/item/weapon/pinpointer/Destroy()
-	active = 0
-	pinpointer_list -= src
+	STOP_PROCESSING(SSfastprocess, src)
 	return ..()
 
-/obj/item/weapon/pinpointer/attack_self()
-	if(!active)
-		active = 1
-		workdisk()
-		usr << "<span class='notice'>You activate the pinpointer.</span>"
-	else
-		active = 0
-		icon_state = "pinoff"
-		usr << "<span class='notice'>You deactivate the pinpointer.</span>"
-
-/obj/item/weapon/pinpointer/proc/scandisk()
-	if(!the_disk)
-		the_disk = locate()
-
-/obj/item/weapon/pinpointer/proc/point_at(atom/target, spawnself = 1)
-	if(!active)
-		return
-	if(!target)
-		icon_state = "pinonnull"
-		return
-
-	var/turf/T = get_turf(target)
-	var/turf/L = get_turf(src)
-
-	if(T.z != L.z)
-		icon_state = "pinonnull"
+/obj/item/weapon/pinpointer/attack_self(mob/living/user)
+	active = !active
+	user.visible_message("<span class='notice'>[user] [active ? "" : "de"]activates their pinpointer.</span>", "<span class='notice'>You [active ? "" : "de"]activate your pinpointer.</span>")
+	playsound(user, 'sound/items/Screwdriver2.ogg', 50, 1)
+	icon_state = "pin[active ? "onnull" : "off"]"
+	if(active)
+		START_PROCESSING(SSfastprocess, src)
 	else
-		setDir(get_dir(L, T))
-		switch(get_dist(L, T))
-			if(-1)
-				icon_state = "pinondirect"
-			if(1 to 8)
-				icon_state = "pinonclose"
-			if(9 to 16)
-				icon_state = "pinonmedium"
-			if(16 to INFINITY)
-				icon_state = "pinonfar"
-	if(spawnself)
-		spawn(5)
-			.()
-
-/obj/item/weapon/pinpointer/proc/workdisk()
-	scandisk()
-	point_at(the_disk, 0)
-	spawn(5)
-		.()
+		target = null //Restarting the pinpointer forces a target reset
+		STOP_PROCESSING(SSfastprocess, src)
 
 /obj/item/weapon/pinpointer/examine(mob/user)
 	..()
+	switch(mode)
+		if(TRACK_NUKE_DISK)
+			user << "Its tracking indicator reads \"nuclear_disk\"."
+		if(TRACK_MALF_AI)
+			user << "Its tracking indicator reads \"01000001 01001001\"."
+		if(TRACK_INFILTRATOR)
+			user << "Its tracking indicator reads \"vasvygengbefuvc\"."

```

Comment:
Rot13 for "infiltratorship", even though it is Syndicate, not Steampunk. Still, nice little touch.